src/tools/illink/Directory.Build.props

```diff
@@ -9,30 +10,16 @@
     <NetCoreAppTestsCurrent>net7.0</NetCoreAppTestsCurrent>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(IsReferenceAssembly)' == 'true' ">
-    <!-- Since .NET 5 reference assemblies are always produced -->
-    <ProduceReferenceAssembly>false</ProduceReferenceAssembly>
     <ProduceOnlyReferenceAssembly>true</ProduceOnlyReferenceAssembly>
     <!-- Used by Arcade to compute OutputPath, IntermediateOutputPath, etc. early in the import chain. -->
     <OutDirName>$(MSBuildProjectName)/ref</OutDirName>
```

Comment:
Should not this align with other names we have in artefacts folder?

/cc @ViktorHofer

Reply:
Yes. We already have infrastructure for
- building reference assemblies (referenceAssemblies.props)
- referencing them from the source project (resolveContract.targets)
- validating them against the implementation (APICompat)
- re-generating them (GenAPI)
in libraries. Let's make sure to use that instead of duplicating it here. @tlakollo if you don't mind, I can help with that as part of this PR.